src/crepe/system/CollisionSystem.cpp

```diff
@@ -1,5 +1,381 @@
+
+#include "api/Event.h"
+#include "api/EventManager.h"
+#include "api/BoxCollider.h"
+#include "api/CircleCollider.h"
+#include "api/Vector2.h"
+#include "api/Rigidbody.h"
+#include "api/Transform.h"
+
+#include "ComponentManager.h"
 #include "CollisionSystem.h"
+#include "Collider.h"
+#include "types.h"
+#include "util/OptionalRef.h"
 
 using namespace crepe;
 
-void CollisionSystem::update() {}
+void CollisionSystem::update() {
+	// Get collider components and keep them seperate
+	ComponentManager & mgr = this->component_manager;
+	std::vector<std::reference_wrapper<BoxCollider>> boxcolliders	= mgr.get_components_by_type<BoxCollider>();
+	std::vector<std::reference_wrapper<CircleCollider>> circlecolliders	= mgr.get_components_by_type<CircleCollider>();
+
+	std::vector<collider_stor> all_colliders;
+	// Add BoxCollider references
+	for (auto& box : boxcolliders) {
+			all_colliders.push_back(collider_stor{box});
+	}
+
+	// Add CircleCollider references
+	for (auto& circle : circlecolliders) {
+			all_colliders.push_back(collider_stor{circle});
+	}
+	
+	// Check between all colliders if there is a collision
+	std::vector<std::pair<CollidedInfoStor,CollidedInfoStor>> collided = check_collisions(all_colliders);
+
+	// For both objects call the collision handler 
+	for (auto& collision_pair : collided) {
+		collision_handler_request(collision_pair.first,collision_pair.second);
+		collision_handler_request(collision_pair.second,collision_pair.first);
+	}
+}
+
+void CollisionSystem::collision_handler_request(CollidedInfoStor& data1,CollidedInfoStor& data2){
+
+	ColliderStorType type = check_collider_type(data1.collider,data2.collider);
+	std::pair<vec2,CollisionSystem::Direction> move_back_data = collision_handler(data1,data2,type);
+
+	OptionalRef<Collider> collider1;
+	OptionalRef<Collider> collider2;
+	switch (type) {
+		case ColliderStorType::BOX_BOX:{
+			collider1 = std::get<std::reference_wrapper<BoxCollider>>(data1.collider);
+			collider2 = std::get<std::reference_wrapper<BoxCollider>>(data2.collider);
+			break;
+		}
+		case ColliderStorType::BOX_CIRCLE:{
+			collider1 = std::get<std::reference_wrapper<BoxCollider>>(data1.collider);
+			collider2 = std::get<std::reference_wrapper<CircleCollider>>(data2.collider);
+			break;
+		}
+		case ColliderStorType::CIRCLE_BOX:{
+			collider1 = std::get<std::reference_wrapper<CircleCollider>>(data1.collider);
+			collider2 = std::get<std::reference_wrapper<BoxCollider>>(data2.collider);
+			break;
+		}
+		case ColliderStorType::CIRCLE_CIRCLE:{
+			collider1 = std::get<std::reference_wrapper<CircleCollider>>(data1.collider);
+			collider2 = std::get<std::reference_wrapper<CircleCollider>>(data2.collider);
+			break;
+		}
+	}
+	
+	// collision info
+	crepe::CollisionSystem::CollisionInfo collision_info{
+            .first={ collider1, data1.transform, data1.rigidbody },
+            .second={ collider2, data2.transform, data2.rigidbody },
+						.move_back_value = move_back_data.first,
```

Comment:
raare spacing

Reply:
zie comment make format is nog niet gedaan